src/shmem_synchronization.h

```diff
@@ -140,16 +158,40 @@ shmem_internal_fence(shmem_ctx_t ctx)
         }                                                               \
     } while(0)
 
+#define SHMEM_SIGNAL_WAIT_UNTIL_BLOCK(var, cond, value, sat_value)      \
+    do {                                                                \
+        uint64_t target_cntr;                                           \
+        int cmpret;                                                     \
+                                                                        \
+        COMP_SIGNAL(cond, SYNC_LOAD(var), value, cmpret, sat_value);    \
+        while (!cmpret) {                                               \
+            target_cntr = shmem_transport_received_cntr_get();          \
+            COMPILER_FENCE();                                           \
+            COMP(cond, SYNC_LOAD(var), value, cmpret);                  \
+            if (cmpret) break;                                          \
```

Comment:
If we break from here, does the `sat_value` get updated with the correct value? We are updating it only in `COMP_SIGNAL`.. Let me know if I am not seeing anything obvious.

Reply:
Yikes, this is definitely a mistake - it should be `COMP_SIGNAL`.  Thanks very much @wrrobin!

I'll merge after you approve this fix and testing finishes.